File: hall_of_fame/gusah009/Java/[PGS]오픈채팅방.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

class Solution {

  public String[] solution(String[] record) {
    Map<String, String> map = new HashMap<>();
    for (String r : record) {
      String[] s = r.split(" ");
      String command = s[0];
      String src = s[1];
      switch (command) {
        case "Enter" -> {
          String dst = s[2];
          map.put(src, dst);
        }
        case "Leave" -> {
        }
        case "Change" -> {
          System.out.println(Arrays.toString(s));
          String dst = s[2];
          map.put(src, dst);
        }
      }
    }

    List<String> answer = new ArrayList<>();
    for (String r : record) {
      String[] s = r.split(" ");
      String command = s[0];
      String member = s[1];
      switch (command) {
        case "Enter" -> {
          answer.add(map.get(member) + "님이 들어왔습니다.");
        }
        case "Leave" -> {
          answer.add(map.get(member) + "님이 나갔습니다.");
        }
      }
    }
    return answer.toArray(String[]::new);
  }
}

File: dajeong-kim/hash/[PGS]오픈채팅방.py
def solution(record):
    # uid를 key, 닉네임을 value로 하는 딕셔너리
    uid_dict = {} 
    for line in record:
        cmd = line.split(" ")
        if cmd[0] != "Leave": #Enter또는 Change일때만 닉네임 값 변경됨
            uid_dict[cmd[1]] = cmd[2]
    
    answer = []
    for line in record:
        cmd = line.split(" ")
        
        if cmd[0] == "Enter":
            #answer.append("%s님이 들어왔습니다." % uid_dict[cmd[1]])
            answer.append(f"{uid_dict[cmd[1]]}님이 들어왔습니다.")
        elif cmd[0] == "Leave":
            #answer.append("%s님이 나갔습니다." % uid_dict[cmd[1]])
            answer.append(f"{uid_dict[cmd[1]]}님이 나갔습니다.")
    
    return answer